src/Compilers/CSharp/Test/Semantic/Semantics/RefExtensionMethodsTests.cs

```diff
@@ -1824,11 +1829,11 @@ public class Program
     public static void Main()
     {
         var obj = new S1 { i = 5 };
-        obj.PrintValue();
+        obj.PrintValue(obj);
     }
 }";
 
-            CompileAndVerify(code, additionalRefs: new[] { SystemCoreRef }, expectedOutput: "54321");
+            CompileAndVerify(code, additionalRefs: new[] { SystemCoreRef }, expectedOutput: "5554535251");
```

Comment:
Please split the test in two (one for mutating and one for not mutating). I cannot compute the expected result.

Reply:
isolated nonmutating test will loop forever. 
The testcase is specifically to show that we handle recursive case correctly, but in nearly identical contexts we are able to handle writeable and readonly case differently. 

I think, while it is a complex test, the complexity serves the purpose here. 

---
In reply to: [139490678](https://github.com/dotnet/roslyn/pull/21982#discussion_r139490678) [](ancestors = 139490678)